navigator_helpers/tasks/evaluation/evaluation.py

```diff
@@ -0,0 +1,764 @@
+                # All unique integer values, potentially an ID field
+                return "Other"
+            return "Numeric"
+
+        if diff_percent >= 0.9 or (diff_percent >= 0.7 and len(non_na_data) <= 50):
+            return "Categorical"
+
+        if space_count / non_na_count > _TEXT_FIELD_AVG_SPACE_COUNT_THRESHOLD:
+            # Count datetime fields as "Other"
+            try:
+                # pd.to_datetime(non_na_data)
+                pd.to_datetime(non_na_data, format="%Y-%m-%d")
+                return "Other"
+            except:
+                pass
+            return "Text"
+
+        return "Other"
+
+    def _get_tfidf_vectors(self, column: Series) -> np.ndarray:
+        vectorizer = TfidfVectorizer().fit_transform(column)
+        return vectorizer.toarray()
+
+    def _get_llm_as_a_judge_prompt(self, natural_language: str, code: str) -> str:
+        rubric = llm_as_a_judge_prompts.general_response_quality_rubric
+        prompt = rubric(
+            natural_language=natural_language,
+            code=code,
+        )
+        return prompt
+
+    def _eval_response_with_llm_as_a_judge(
+        self,
+        natural_language: str,
+        code: str,
+        **kwargs,
+    ) -> dict:
+        prompt = self._get_llm_as_a_judge_prompt(
+            natural_language=natural_language, code=code, **kwargs
+        )
+        scores = {}
+        response = utils.parse_json_str(self.llm_suite.judge_generate(prompt)) or {}
+        for k, v in response.items():
+            scores[f"{k}_score"] = v["score"]
+            scores[f"{k}_reason"] = v["reasoning"]
+        return scores
+
+    def row_uniqueness(self):
+        """
+        Evaluation test for brute force row uniqueness & semantic uniqueness using TF-IDF and cosine similarity.
+        Returns:
+            - Percentage of rows that are unique
+            - Percentage of rows that are semantically unique (fuzzy match)
+            - IDs of rows that are not unique
+            - IDs of rows that are not semantically unique
+        Example for semantic duplicates:
+            - Prompt 1 : "How to implement k-means clustering in Python?"
+            - Prompt 2 : "Can you implement k-means clustering in Python?"
+        """
+        # TODO: Try out https://github.com/dleemiller/WordLlama for semantic uniqueness
+
+        total_rows = len(self.dataset)
+
+        # Brute Force row uniqueness by detecting duplicates
+        unique_rows = self.dataset.drop_duplicates()
+        non_unique_ids = self.dataset[self.dataset.duplicated()].index.tolist()
+        percent_unique = len(unique_rows) / total_rows * 100
+
+        # Semantic uniqueness using cosine similarity and TF-IDF
+        text_columns = self.dataset.select_dtypes(include=[object])
+        # text_columns = self.dataset.select_dtypes(include=[object, 'string'])
+        concatenated_text = text_columns.apply(
```

Comment:
Should we sort the column names in order to make sure the train and synthetic text columns are in the same order before merging them together?